.github/workflows/documentation.yml

```diff
@@ -11,6 +11,9 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
+permissions:
+  contents: read
```

Comment:
This workflow pushes to gh-pages branch. Will this break the workflow?

Reply:
Should be fine, as you have specific permission at the job level: https://github.com/ada-url/ada/blob/3113fafbc16a1de100b7919d35a2b6699966e8d8/.github/workflows/documentation.yml#L19